### PR TITLE
Remaining database cleanup, tests now passing.

### PR DESCRIPTION
**Pull request by JackUrb**
**Patch description**

**Testing steps**
```bash
>> pytest light
...............................................sssssssssssss.........ssssssssss..s..s..s..s....................................................................................................... [ 88%]
.........................                                                                                                                                                                          [100%]
============================================================================================ warnings summary ============================================================================================
/Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/numexpr/expressions.py:21
/Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/numexpr/expressions.py:21
  /Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/numexpr/expressions.py:21: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    _np_version_forbids_neg_powint = LooseVersion(numpy.__version__) >= LooseVersion('1.12.0b1')

/Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/scipy/special/orthogonal.py:79
/Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/scipy/special/orthogonal.py:79
  /Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/scipy/special/orthogonal.py:79: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    from numpy import (exp, inf, pi, sqrt, floor, sin, cos, around, int,

light/world/tests/test_agent_death.py::TestInteractionLoggers::test_run
  /Users/jju/miniconda3/envs/py38/lib/python3.8/asyncio/events.py:81: RuntimeWarning: coroutine 'BattleRoyaleSoul.observe_event' was never awaited
    self._context.run(self._callback, *self._args)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======================================================================================== short test summary info =========================================================================================
SKIPPED [8] light/graph/builders/tests/test_StarSpaceBuilder.py: Need to update starspace builders to use Model Pool
SKIPPED [5] light/graph/builders/tests/test_StarspaceNeighborBuilder.py: Need to update starspace builders to use Model Pool
SKIPPED [7] light/graph/builders/tests/test_user.py: Need to update starspace builders to use Model Pool
SKIPPED [3] /Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/_pytest/unittest.py:258: Skip GraphEventTests tests, it's a base class
SKIPPED [1] light/graph/tests/test_events.py:230: No valid events for <class 'light.graph.events.graph_events.SayEvent'>
SKIPPED [1] light/graph/tests/test_events.py:230: No valid events for <class 'light.graph.events.graph_events.ShoutEvent'>
SKIPPED [1] light/graph/tests/test_events.py:230: No valid events for <class 'light.graph.events.graph_events.WhisperEvent'>
SKIPPED [1] light/graph/tests/test_events.py:230: No valid events for <class 'light.graph.events.graph_events.TellEvent'>
192 passed, 27 skipped, 5 warnings in 14.29s
```

### CODE REVIEW

File: light/data_model/db/environment.py [hunk 1]
```diff
@@ -70,13 +79,17 @@ class DBNameKey(HasDBIDMixin):
     id and a name
     """
 
-    db_id: str = Column(String(ID_STRING_LENGTH), primary_key=True)  # type: ignore
-    name: str = Column(String(BASE_NAME_LENGTH_CAP), nullable=False, index=True, unique=True)  # type: ignore
-    status: DBStatus = Column(Enum(DBStatus), nullable=False, index=True)  # type: ignore
-    split: DBSplitType = Column(Enum(DBSplitType), nullable=False, index=True)  # type: ignore
+    db_id: Mapped[str] = mapped_column(String(ID_STRING_LENGTH), primary_key=True)
```
Comment (mojtaba-komeili) [inline light/data_model/db/environment.py:82]: Could this be related the SQLAlchemy issue I ran into today? It was complaining about the mapped columns too.

File: light/data_model/db/episodes.py [hunk 2]
```diff
@@ -29,14 +39,17 @@
...
 
-SQLBase = declarative_base()
 FILE_PATH_KEY = "episodes"
 ID_STRING_LENGTH = 40
 USR_KEY = DBPlayer.ID_PREFIX
 MAX_WILD_MODEL_LEN = 200
 MAX_WILD_CHOICE_LEN = 100
 
 
+class SQLBase(DeclarativeBase):
```
Comment (mojtaba-komeili) [inline light/data_model/db/episodes.py:49]: Just a question, why not using `DeclarativeBase` directly? Do you plan to extend it later, or just having a more descriptive name?

File: requirements.txt [hunk 3]
```diff
@@ -10,4 +10,4 @@ pyzmq>=19.0.1
 tqdm>=4.48.0
 hydra-core>=1.2.0
 mephisto>=1.0.3
-SQLAlchemy>=1.4.36
+SQLAlchemy>=2.0.7
```
Comment (mojtaba-komeili) [inline requirements.txt:13]: Noted! 
I changed back my requirements to this in my PR and test with that.

**Review by mojtaba-komeili — APPROVED** [review]
Had a quick look and it makes sense. Was it mostly about the SQLAlchemy deprecations?

File: light/data_model/db/environment.py (hunk 1 above)
Comment (JackUrb) [reply light/data_model/db/environment.py:82]: Yup this was the cause of all of that.

File: light/data_model/db/episodes.py (hunk 2 above)
Comment (JackUrb) [reply light/data_model/db/episodes.py:49]: You're not allowed to extend `DeclarativeBase` directly due to some SQLAlchemy metaclass magic

**Comment by JackUrb** [comment]
Mostly, with some very minor logic edits resolving bugs.

**Pull request merged**